lib/web3/contract.js

```diff
@@ -57,7 +56,7 @@ var addFunctionsToContract = function (contract, abi) {
     abi.filter(function (json) {
         return json.type === 'function';
     }).map(function (json) {
-        return new SolidityFunction(json, contract.address);
+        return new SolidityFunction(contract._web3, json, contract.address);
```

Comment:
I might be wrong, but I think that all the things that we call `solidity` in source code, **were** initially part only of solidity ABI specification. They are not a part of evm, nor of yellowpaper. Certain implementations of other languages (eg. LLL) adopted this ABI. Am I right @chriseth ?